modules/raft-api/src/main/java/org/apache/ignite/internal/raft/Command.java

```diff
@@ -17,10 +17,27 @@
 
 package org.apache.ignite.internal.raft;
 
+import org.apache.ignite.internal.hlc.HybridTimestamp;
 import org.apache.ignite.internal.network.NetworkMessage;
+import org.jetbrains.annotations.Nullable;
 
 /**
  * A marker interface for replication group command.
  */
 public interface Command extends NetworkMessage {
+    /**
+     * This is called before a command is submitted to replication pipeline.
+     *
+     * @param safeTs Safe timestamp.
+     */
+    default void patch(HybridTimestamp safeTs) {}
```

Comment:
Let's keep it generic. This is the same as current before apply patching works and helps to avoid casting.
I think we can get rid of Read/WriteCommand interfaces in the future.